lib/application_json.rb

```diff
@@ -300,38 +300,24 @@ def reference_attributes
     ]
   end
 
-  def offer_json
-    JSON.pretty_generate(json_data['offer'])
+  def offer_conditions_json
+    JSON.pretty_generate(conditions: json_data['offer_conditions'])
   end
 
-  def offer_creation_json
-    course_details = json_data['offer']['course'].dup # we are going to mutate this :/
-    course_details.delete('description')
-    JSON.pretty_generate(
-      json_data['offer'].merge('course' => course_details)
-    )
-  end
-
-  def offer_attributes
+  def offer_conditions_attributes
     [
-      {
-        name: 'course',
-        type: link_to_resource_definition('Course'),
-        description: 'Course details'
-      },
-      {
-        name: 'date',
-        type: 'string',
-        description: 'The offered entry date in YYYY-MM-DD format'
-      },
       {
         name: 'conditions',
```

Comment:
Should this be `offer_conditions` as well?

Reply:
Ah yes, good spot! I shall update that. 👌

Reply:
After a discussion IRL about this, we decided to change it back to `offer` as this would be easier to change.